### DISCUSSION TITLE
Default and empty values

### DISCUSSION THREAD

**Discussion by gsicvj** [tinacms/tinacms#3985] [Help]
Hi all, inside a larger codebase it's expensive to refactor code in the new mental model TinaCMS adopted.
I wasn't sure whether to write a "Help" or "Idea" type of discussion, but it's more of a problem for my project than idea.

2 ways of handling empty values:
- You keep them (Forestry)
- You delete the property (TinaCMS)

This article explains hacks to the problem: https://pre-beta.tina.io/guides/gatsby/git/empty-field-errors/
But trying the parse method doesn't actually work as TinaCMS still erases an empty array.

I believe there was also a discussion in another thread on github, but I can't seem to find that, in which someone mentions having empty values is wrong, but that's maybe a topic for another discussion.

I would be okay with either if I started a new project, but maintaining an existing project results in many hours spent trying to deploy and talking with editors why the build fails.

It would be nice if there was a switch to turn of how TinaCMS handles the data deletion for empty fields.
And even more if there was a switch to turn on having empty fields again and let developers decide what they want to go with.

**Comment by jamespohalloran** [comment]
Hey, do you mind explaining the behaviour you're seeing a bit more? Also, what version of tinacms are you on?
The doc you're referencing is on the pre-beta docs, I don't recognize that described behaviour, and I wouldn't think it applied to the latest version of tinacms, but maybe I'm misunderstanding. Tinacms shouldn't be deleting fields with empty values.
